### PR TITLE
SW-4394 Batch Details tab

### PR DESCRIPTION
**Pull request by constanzauanini**
- Show all batch's details in Details tab, including photos
- Open Edit Details modal when clicking edit button
- Allow editing batch details in modal
- Allow adding/removing photos from details modal

Screenshot 2023-11-06 at 15 54 23

Screenshot 2023-11-06 at 15 58 31

### CODE REVIEW

File: src/components/InventoryV2/BatchDetailsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,402 @@
+import { useEffect, useState, useMemo } from 'react';
```
Comment (constanzauanini) [inline src/components/InventoryV2/BatchDetailsModal.tsx:1]: This modal is based on v1, but adding some new fields and all photos management

File: src/components/InventoryV2/BatchDetails.tsx [hunk 2]
```diff
@@ -1,15 +1,133 @@
...
+
+  const overviewCardData = [
+    {
+      title: strings.GERMINATING_QUANTITY,
+      data: batch.germinatingQuantity,
+    },
+    {
+      title: '',
+      data: '',
+    },
```
Comment (charlie-tf) [inline src/components/InventoryV2/BatchDetails.tsx:33]: Just for space? Will the space be okay when in mobile?

File: src/components/InventoryV2/BatchDetailsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    onChange(id, date);
+  };
+
+  const marginTop = {
+    marginTop: theme.spacing(2),
+  };
+
+  return (
+    <>
+      {record && (
```
Comment (charlie-tf) [inline src/components/InventoryV2/BatchDetailsModal.tsx:213]: This seems okay. An alternative would be to do
```typescript
return (
  {record
    ? <DialogBox>
      ...
      </DialogBox>
    : null}
);
```
Which would avoid returning an empty fragment. You'd have to change the return type to `JSX.Element | null` though.

**Review by charlie-tf — COMMENTED** [review]
Looks good! Main question is about the empty `overviewCardData` member

File: src/components/InventoryV2/BatchDetails.tsx (hunk 2 above)
Comment (constanzauanini) [reply src/components/InventoryV2/BatchDetails.tsx:33]: Oh, you are right, I didn't test it in mobile. Let me check

File: src/components/InventoryV2/BatchDetails.tsx [hunk 4]
```diff
@@ -26,10 +26,7 @@ export default function BatchDetails({ batch, onUpdate }: BatchDetailsProps): JS
     {
       title: strings.GERMINATING_QUANTITY,
       data: batch.germinatingQuantity,
-    },
-    {
-      title: '',
-      data: '',
+      fullWidth: true,
```
Comment (charlie-tf) [inline src/components/InventoryV2/BatchDetails.tsx:29]: Nice update!

**Review by charlie-tf — APPROVED** [review]
LGTM

**Pull request merged**